### ISSUE TITLE
PayPal::SDK::REST::ErrorHash.convert is not working correctly on nested hashes

### ISSUE THREAD

**Issue by hkmaly** [paypal/PayPal-Ruby-SDK#257]
### General Information

- Mode (Sandbox/Live): Live
- PayPal-Debug-ID(s) (from any logs): N/A  (well, noticed this when investigating 13a48c8bc07b0 but that one doesn't contains nested hash)
- Version of Ruby used: 1.9.3
- SDK Version: 1.6.0

### Issue Description

PayPal::SDK::REST::ErrorHash.convert({:a=>1, :b=>2, :inside=>{:c=>1, :d=>2}}) throws error.
```
/var/lib/gems/1.9.1/gems/paypal-sdk-rest-1.6.0/lib/paypal-sdk/rest/error_hash.rb:7:in `convert': undefined method `each' for -2649451557909952569:Fixnum (NoMethodError)
        from /var/lib/gems/1.9.1/gems/paypal-sdk-rest-1.6.0/lib/paypal-sdk/rest/error_hash.rb:16:in `[]='
        from /var/lib/gems/1.9.1/gems/paypal-sdk-rest-1.6.0/lib/paypal-sdk/rest/error_hash.rb:8:in `block in convert'
        from /var/lib/gems/1.9.1/gems/paypal-sdk-rest-1.6.0/lib/paypal-sdk/rest/error_hash.rb:7:in `each'
        from /var/lib/gems/1.9.1/gems/paypal-sdk-rest-1.6.0/lib/paypal-sdk/rest/error_hash.rb:7:in `convert'
```
Looking at the code, 
```
         def []=(key, value)
          value =
            if value.is_a? Hash
              ErrorHash.convert(hash)
            elsif value.is_a? Array and value[0].is_a? Hash
              value.map{|array_value| ErrorHash.convert(array_value) }
            else
              value
            end
          super(key, value)
        end
```
looks like it REALLY shoud be ErrorHash.convert(value) instead of ErrorHash.convert(hash)

Note that this is likely cause of #240 and you should revert that flattening "fix" and fix this issue instead. And stop blaming ruby core.

**Comment by hkmaly** [comment]
Hello,
any progress on this?

**Comment by hkmaly** [comment]
Hello,
any progress on this?

**Comment by pp-randy** [comment]
@hkmaly I have escalated this issue internally for further review, and I apologize for the delay.  

Internal Tracking: BTDXCLTS-260

**Comment by pp-randy** [comment]
@hkmaly a change was merged in https://github.com/paypal/PayPal-Ruby-SDK/issues/280, did you want to take a look?

**Comment by hkmaly** [comment]
Did you also reverted #240 ? That "fix" is very bad idea and I believe with fixing this it will no longer be needed (I assume your tests will confirm that).

**Comment by pp-randy** [comment]
I don't see that https://github.com/paypal/PayPal-Ruby-SDK/pull/240 was reverted.  Are you still seeing problems though now that https://github.com/paypal/PayPal-Ruby-SDK/pull/280 was merged in?

**Comment by hkmaly** [comment]
As I commented, I was investigating something which turned out to be unrelated (as the hash was already flat in that case, specifically I'm not getting avs/cvv responses from failed payment which seem to be reason for non-flat hash) when I found this problem.

I'm not sure what exactly was triggering #240, but flattening nested hashes is terrible idea (two fields can overwrite each other if they have same key) and you shouldn't ever do it - and I believe now it wouldn't be needed, because #280 would fix #234 better than #240.

(Also, I'm not sure how to install gem directly from git, without being tagged. I'm monkey-patching this. So, sorry, I'm not the best one for testing if it works.)
